src/concrete/ml/quantization/quantized_module.py

```diff
@@ -271,6 +282,26 @@ def post_processing(
         """
         return values[0] if len(values) == 1 else values
 
+    def pre_processing(self, *values: numpy.ndarray) -> Tuple[numpy.ndarray, ...]:
+        """Apply pre-processing to the input values.
+
+        For quantized modules, this method applies pre-processing if a NumpyModule exists,
+        or creates one if ONNX preprocessing is defined but the NumpyModule hasn't been created yet.
+
+        Args:
+            values (numpy.ndarray): The input values to pre-process.
+
+        Returns:
+            Tuple[numpy.ndarray, ...]: The pre-processed values.
+        """
+        if self._preprocessing_numpy is None and self._onnx_preprocessing is not None:
```

Comment:
so basically, are we expecting someone to run `pre_processing` multiple times in a row ? : 
- if not, then there's no need to have a `self._preprocessing_numpy` attibute anymore, we simple build and return the numpy module directly)
- if yes, then : 
    - we could keep building this numpy on the fly if it's quick 
    - if it's too slow, then why not instead directly build the numpy module in the QM's init and only handle `_preprocessing_numpy` (no need to keep `_onnx_preprocessing`) ? and for serialization you could dump the onnx processing (you could store `model` in `NumpyModule` if needed) and load back the numpy module in load.

it's just that we probably want to avoid adding more attributes to important classes like QMs if it's avoidable !

Reply:
Well everytime a call to forward is done the pre-processing must be applied yes. We build a numpy module from onnx I don't think it's good practice to build it everytime.

Agreed to have it built in init and load time. Also agreed to remove the _onnx_preprocessing attribute and only use the _preprocessing_numpy.

Reply:
> side note : here QM's _preprocessing_numpy is a NumpyModule instance, but a NumpyModule instance also has a numpy_preprocessing attribute that is a callable. I feel like we could provide better names in order to avoid confusion !

I updated to `_preprocessing_module`. I see what you mean. Also it could be bad that a onnx_preprocessing gets its own numpy_preprocessing... Not sure how to handle that.